cmd/telemetrygen/internal/metrics/config.go

```diff
@@ -0,0 +1,138 @@
+// Copyright The OpenTelemetry Authors
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/pflag"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+)
+
+var (
+	errFormatOTLPAttributes       = fmt.Errorf("value should be of the format key=\"value\"")
+	errDoubleQuotesOTLPAttributes = fmt.Errorf("value should be a string wrapped in double quotes")
+)
+
+// Config describes the test scenario.
+type Config struct {
+	WorkerCount       int
+	NumMetrics        int
+	Rate              int64
+	TotalDuration     time.Duration
+	ReportingInterval time.Duration
+
+	// OTLP config
+	Endpoint           string
+	Insecure           bool
+	UseHTTP            bool
+	Headers            KeyValue
+	ResourceAttributes KeyValue
+}
+
+type KeyValue map[string]string
```

Comment:
Instead of copying this struct definition, can we move it to `internal/common` and share it between metrics and traces?